### PR TITLE
[RFC] Lazyload Tim

### PR DESCRIPTION
**Pull request by Magellol**
Introduce a lazyloader module, loaded over ES modules that would probably only work in Chrome.
Also, have you seen Jared's new lips? Gorgeous.

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -1,2 +1,7 @@
 <title>C'est Bones!</title>
-<img src="tim.jpg">
+
+<script type="module">
```
Comment (Magellol) [inline index.html]: What is our browser support? I found it hard to make implementation decisions when I don't have this data on hand.

File: lazyloader.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+/**
+ * This is 100% not cross-browser.
+ * We like to use cutting edge technologies in the organization.
+ */
+export const load = source => {
+  const img = new Image();
+  img.src = source;
+
+  return img.decode().then(() => img);
```
Comment (Magellol) [inline lazyloader.js:9]: Ref: https://html.spec.whatwg.org/multipage/embedded-content.html#dom-img-decode

This is the best way to handle loading images asynchronously as the decoding doesn't happen in the main thread but instead a worker is used for it.

File: index.html (hunk 1 above)
Comment (kirillzakharov) [reply index.html]: Currently, we're supporting all browsers. Including IE6.

**Comment by kirillzakharov** [comment]
>Lazyload TIm

**Comment by kirillzakharov** [comment]
Fuck you. Fuck you.

File: lazyloader.js (hunk 2 above)
Comment (kirillzakharov) [reply lazyloader.js:9]: Thoughts on a `nomodule` fallback?

Ref: https://html.spec.whatwg.org/multipage/scripting.html#attr-script-nomodule

File: index.html (hunk 1 above)
Comment (lukechesser) [reply index.html]: Tim needs to be seen on as many platforms as possible.

Also last time I checked, Tim uses Android. So who knows what version he's on. Could be 2009.

File: lazyloader.js (hunk 2 above)
Comment (lukechesser) [reply lazyloader.js:9]: Nit: this could benefit from the cache, state, view, dispatch pattern made popular in 2011.

Comment (Magellol) [reply lazyloader.js:9]: @kirillzakharov You're 100% right. We need a fallback to ancient browsers that aren't capable to keep up with the spec. I will provide a fix for it.

Comment (Magellol) [reply lazyloader.js:9]: Makes sense. Or we can use react, redux, recompose, reselect, jest and mongodb.

**Review by lukechesser — CHANGES_REQUESTED** [review]
Needs fallback so that the whole world can experience bones.

**Comment by Magellol** [comment]
@lukechesser Please see 20359d8 for the requested changes. I have taken the liberty of doing some structural changes in the project.

**Review by kirillzakharov — APPROVED** [review]

**Review by lukechesser — APPROVED** [review]
`fuckingSlowImage`

**Comment by lukechesser** [comment]
Please make sure to QA your PRs though. Continuous testing is not currently setup on this project.

**Comment by Magellol** [comment]
Thanks for the review. I did QA'd it, I'm not a savage.

**Pull request merged**